### ISSUE TITLE
Tower 3.6 install re-run fails when postgres10 is already setup (fires 9.6 upgrade play)

### ISSUE THREAD

**Issue by ffirg** [ansible/awx#5357]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
If you re-run the setup.sh installer after it's already done a fresh new external install of the DB on postgres10, the installer will fail as it's calling the roles/packages_el/tasks/upgrade_postgres.yml role to upgrade Postgres.

It's also referring to 9.6 when clearly doing 10.x DB tasks (see output below)

##### ENVIRONMENT
* Tower version: 3.6.0
* Tower install method: normal VM method in AWS
* Ansible version:  2.8.4
* Operating System: RHEL 8.1
* Web Browser: Chrome 78.0.3904.97

##### STEPS TO REPRODUCE

Run a fresh setup.sh install setting up a brand new postgres10 database node. Then re-run it and it will fail.

##### EXPECTED RESULTS

setup.sh should not fire on upgrade postgres (TASK [packages_el : Upgrade Postgres 10 [RHEL 8]]) as it's already detected nothing to do in the previous task:

TASK [packages_el : Install Postgresql 10 [RHEL8]] **********************************************
ok: [ec2-xxx.eu-west-2.compute.amazonaws.com] => {"changed": false, "msg": "Nothing to do", "rc": 0, "results": []}

##### ACTUAL RESULTS

TASK [packages_el : Install Postgresql [RHEL8]] *************************************************
skipping: [ec2-xxx.eu-west-2.compute.amazonaws.com] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [packages_el : include] ********************************************************************
included: /root/ansible-tower-setup-3.6.0-1/roles/packages_el/tasks/upgrade_postgres.yml for ec2-xxx.eu-west-2.compute.amazonaws.com

TASK [packages_el : Stop postgres 9.6] **********************************************************
changed: [ec2-xxx.eu-west-2.compute.amazonaws.com] => {"changed": true, "enabled": false, "name": "postgresql", "state": "stopped", "status": {"ActiveEnterTimestamp": "Wed 2019-11-20 10:38:15 UTC", "ActiveEnterTimestampMonotonic": "2465064002", "ActiveExitTimestamp": "Wed 2019-11-20 10:38:14 UTC", "ActiveExitTimestampMonotonic": "2464909156", "ActiveState": "active", "After": "sysinit.target basic.target system.slice systemd-journald.socket network.target", "AllowIsolate": "no", "AmbientCapabilities": "", "AssertResult": "yes", "AssertTimestamp": "Wed 2019-11-20 10:38:15 UTC", "AssertTimestampMonotonic": "2464927101", "Before": "multi-user.target shutdown.target", "BlockIOAccounting": "no", "BlockIOWeight": "[not set]", "CPUAccounting": "no", "CPUQuotaPerSecUSec": "infinity", "CPUSchedulingPolicy": "0", "CPUSchedulingPriority": "0", "CPUSchedulingResetOnFork": "no", "CPUShares": "[not set]", "CPUUsageNSec": "[not set]", "CPUWeight": "[not set]", "CacheDirectoryMode": "0755", "CanIsolate": "no", "CanReload": "yes", "CanStart": "yes", "CanStop": "yes", "CapabilityBoundingSet": "cap_chown cap_dac_override cap_dac_read_search cap_fowner cap_fsetid cap_kill cap_setgid cap_setuid cap_setpcap cap_linux_immutable cap_net_bind_service cap_net_broadcast cap_net_admin cap_net_raw cap_ipc_lock cap_ipc_owner cap_sys_module cap_sys_rawio cap_sys_chroot cap_sys_ptrace cap_sys_pacct cap_sys_admin cap_sys_boot cap_sys_nice cap_sys_resource cap_sys_time cap_sys_tty_config cap_mknod cap_lease cap_audit_write cap_audit_control cap_setfcap cap_mac_override cap_mac_admin cap_syslog cap_wake_alarm cap_block_suspend", "CollectMode": "inactive", "ConditionResult": "yes", "ConditionTimestamp": "Wed 2019-11-20 10:38:15 UTC", "ConditionTimestampMonotonic": "2464927100", "ConfigurationDirectoryMode": "0755", "Conflicts": "shutdown.target", "ControlGroup": "/system.slice/postgresql.service", "ControlPID": "0", "DefaultDependencies": "yes", "Delegate": "no", "Description": "PostgreSQL database server", "DevicePolicy": "auto", "DynamicUser": "no", "Environment": "PG_OOM_ADJUST_FILE=/proc/self/oom_score_adj PG_OOM_ADJUST_VALUE=0 PGDATA=/var/lib/pgsql/data", "ExecMainCode": "0", "ExecMainExitTimestampMonotonic": "0", "ExecMainPID": "22244", "ExecMainStartTimestamp": "Wed 2019-11-20 10:38:15 UTC", "ExecMainStartTimestampMonotonic": "2464934966", "ExecMainStatus": "0", "ExecReload": "{ path=/bin/kill ; argv[]=/bin/kill -HUP $MAINPID ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }", "ExecStart": "{ path=/usr/bin/postmaster ; argv[]=/usr/bin/postmaster -D ${PGDATA} ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }", "ExecStartPre": "{ path=/usr/libexec/postgresql-check-db-dir ; argv[]=/usr/libexec/postgresql-check-db-dir postgresql ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }", "FailureAction": "none", "FileDescriptorStoreMax": "0", "FragmentPath": "/usr/lib/systemd/system/postgresql.service", "GID": "26", "Group": "postgres", "GuessMainPID": "yes", "IOAccounting": "no", "IOSchedulingClass": "0", "IOSchedulingPriority": "0", "IOWeight": "[not set]", "IPAccounting": "no", "IPEgressBytes": "18446744073709551615", "IPEgressPackets": "18446744073709551615", "IPIngressBytes": "18446744073709551615", "IPIngressPackets": "18446744073709551615", "Id": "postgresql.service", "IgnoreOnIsolate": "no", "IgnoreSIGPIPE": "yes", "InactiveEnterTimestamp": "Wed 2019-11-20 10:38:15 UTC", "InactiveEnterTimestampMonotonic": "2464926421", "InactiveExitTimestamp": "Wed 2019-11-20 10:38:15 UTC", "InactiveExitTimestampMonotonic": "2464927951", "InvocationID": "9ad1ff8e54154e7fa84a08a4b2cda7c7", "JobRunningTimeoutUSec": "infinity", "JobTimeoutAction": "none", "JobTimeoutUSec": "infinity", "KeyringMode": "private", "KillMode": "mixed", "KillSignal": "2", "LimitAS": "infinity", "LimitASSoft": "infinity", "LimitCORE": "infinity", "LimitCORESoft": "infinity", "LimitCPU": "infinity", "LimitCPUSoft": "infinity", "LimitDATA": "infinity", "LimitDATASoft": "infinity", "LimitFSIZE": "infinity", "LimitFSIZESoft": "infinity", "LimitLOCKS": "infinity", "LimitLOCKSSoft": "infinity", "LimitMEMLOCK": "16777216", "LimitMEMLOCKSoft": "16777216", "LimitMSGQUEUE": "819200", "LimitMSGQUEUESoft": "819200", "LimitNICE": "0", "LimitNICESoft": "0", "LimitNOFILE": "4096", "LimitNOFILESoft": "1024", "LimitNPROC": "63346", "LimitNPROCSoft": "63346", "LimitRSS": "infinity", "LimitRSSSoft": "infinity", "LimitRTPRIO": "0", "LimitRTPRIOSoft": "0", "LimitRTTIME": "infinity", "LimitRTTIMESoft": "infinity", "LimitSIGPENDING": "63346", "LimitSIGPENDINGSoft": "63346", "LimitSTACK": "infinity", "LimitSTACKSoft": "8388608", "LoadState": "loaded", "LockPersonality": "no", "LogLevelMax": "-1", "LogsDirectoryMode": "0755", "MainPID": "22244", "MemoryAccounting": "yes", "MemoryCurrent": "184799232", "MemoryDenyWriteExecute": "no", "MemoryHigh": "infinity", "MemoryLimit": "infinity", "MemoryLow": "0", "MemoryMax": "infinity", "MemorySwapMax": "infinity", "MountAPIVFS": "no", "MountFlags": "", "NFileDescriptorStore": "0", "NRestarts": "0", "Names": "postgresql.service", "NeedDaemonReload": "no", "Nice": "0", "NoNewPrivileges": "no", "NonBlocking": "no", "NotifyAccess": "main", "OOMScoreAdjust": "-1000", "OnFailureJobMode": "replace", "PermissionsStartOnly": "no", "Perpetual": "no", "PrivateDevices": "no", "PrivateMounts": "no", "PrivateNetwork": "no", "PrivateTmp": "no", "PrivateUsers": "no", "ProtectControlGroups": "no", "ProtectHome": "no", "ProtectKernelModules": "no", "ProtectKernelTunables": "no", "ProtectSystem": "no", "RefuseManualStart": "no", "RefuseManualStop": "no", "RemainAfterExit": "no", "RemoveIPC": "no", "Requires": "sysinit.target system.slice", "Restart": "no", "RestartUSec": "100ms", "RestrictNamespaces": "no", "RestrictRealtime": "no", "Result": "success", "RootDirectoryStartOnly": "no", "RuntimeDirectoryMode": "0755", "RuntimeDirectoryPreserve": "no", "RuntimeMaxUSec": "infinity", "SameProcessGroup": "no", "SecureBits": "0", "SendSIGHUP": "no", "SendSIGKILL": "yes", "Slice": "system.slice", "StandardError": "inherit", "StandardInput": "null", "StandardInputData": "", "StandardOutput": "journal", "StartLimitAction": "none", "StartLimitBurst": "5", "StartLimitIntervalUSec": "10s", "StartupBlockIOWeight": "[not set]", "StartupCPUShares": "[not set]", "StartupCPUWeight": "[not set]", "StartupIOWeight": "[not set]", "StateChangeTimestamp": "Wed 2019-11-20 10:38:15 UTC", "StateChangeTimestampMonotonic": "2465064002", "StateDirectoryMode": "0755", "StatusErrno": "0", "StopWhenUnneeded": "no", "SubState": "running", "SuccessAction": "none", "SyslogFacility": "3", "SyslogLevel": "6", "SyslogLevelPrefix": "yes", "SyslogPriority": "30", "SystemCallErrorNumber": "0", "TTYReset": "no", "TTYVHangup": "no", "TTYVTDisallocate": "no", "TasksAccounting": "yes", "TasksCurrent": "14", "TasksMax": "26213", "TimeoutStartUSec": "infinity", "TimeoutStopUSec": "infinity", "TimerSlackNSec": "50000", "Transient": "no", "Type": "notify", "UID": "26", "UMask": "0022", "UnitFilePreset": "disabled", "UnitFileState": "enabled", "User": "postgres", "UtmpMode": "init", "WantedBy": "multi-user.target", "WatchdogTimestamp": "Wed 2019-11-20 10:38:15 UTC", "WatchdogTimestampMonotonic": "2465063998", "WatchdogUSec": "0"}}

TASK [packages_el : init postgresql (el-only)] **************************************************
ok: [ec2-xxx.eu-west-2.compute.amazonaws.com] => {"changed": false, "cmd": ["/usr/bin/postgresql-setup", "initdb"], "rc": 0, "stdout": "skipped, since /var/lib/pgsql/data/PG_VERSION exists", "stdout_lines": ["skipped, since /var/lib/pgsql/data/PG_VERSION exists"]}

TASK [packages_el : pg_upgrade [RHEL 7]] ********************************************************
skipping: [ec2-xxx.eu-west-2.compute.amazonaws.com] => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}

TASK [packages_el : Uninstall Postgresql 9.6 [RHEL 8]] ******************************************
changed: [ec2-xxx.eu-west-2.compute.amazonaws.com] => {"changed": true, "cmd": "dnf module reset -y 'postgresql:10'\n", "delta": "0:00:01.685741", "end": "2019-11-20 10:50:31.442827", "rc": 0, "start": "2019-11-20 10:50:29.757086", "stderr": "", "stderr_lines": [], "stdout": "Last metadata expiration check: 0:42:35 ago on Wed 20 Nov 2019 10:07:55 UTC.\nOnly module name is required. Ignoring unneeded information in argument: 'postgresql:10'\nDependencies resolved.\n================================================================================\n Package           Architecture     Version             Repository         Size\n================================================================================\nDisabling module profiles:\n postgresql/server                                                             \nResetting modules:\n postgresql                                                                    \n\nTransaction Summary\n================================================================================\n\nComplete!", "stdout_lines": ["Last metadata expiration check: 0:42:35 ago on Wed 20 Nov 2019 10:07:55 UTC.", "Only module name is required. Ignoring unneeded information in argument: 'postgresql:10'", "Dependencies resolved.", "================================================================================", " Package           Architecture     Version             Repository         Size", "================================================================================", "Disabling module profiles:", " postgresql/server                                                             ", "Resetting modules:", " postgresql                                                                    ", "", "Transaction Summary", "================================================================================", "", "Complete!"]}

TASK [packages_el : Install Postgresql Upgrade packages [RHEL 8]] *******************************
[DEPRECATION WARNING]: Invoking "yum" only once while using a loop via squash_actions is
deprecated. Instead of using a loop to supply multiple items and specifying `name: "{{ item
}}"`, please use `name: ['postgresql-upgrade96']` and remove the loop. This feature will be
removed in version 2.11. Deprecation warnings can be disabled by setting
deprecation_warnings=False in ansible.cfg.
changed: [ec2-xxx.eu-west-2.compute.amazonaws.com] => (item=['postgresql-upgrade96']) => {"ansible_loop_var": "item", "changed": true, "item": ["postgresql-upgrade96"], "msg": "", "rc": 0, "results": ["Installed: postgresql-upgrade96", "Installed: postgresql-upgrade96-9.6.15-2.el8at.x86_64"]}

TASK [packages_el : Enable postgresql 10 dnf modules] *******************************************
changed: [ec2-xxx.eu-west-2.compute.amazonaws.com] => {"changed": true, "cmd": "dnf module -y enable 'postgresql:10'\n", "delta": "0:00:01.991725", "end": "2019-11-20 10:50:36.698377", "rc": 0, "start": "2019-11-20 10:50:34.706652", "stderr": "", "stderr_lines": [], "stdout": "Last metadata expiration check: 0:42:40 ago on Wed 20 Nov 2019 10:07:55 UTC.\nDependencies resolved.\n================================================================================\n Package           Architecture     Version             Repository         Size\n================================================================================\nEnabling module streams:\n postgresql                         10                                         \n\nTransaction Summary\n================================================================================\n\nComplete!", "stdout_lines": ["Last metadata expiration check: 0:42:40 ago on Wed 20 Nov 2019 10:07:55 UTC.", "Dependencies resolved.", "================================================================================", " Package           Architecture     Version             Repository         Size", "================================================================================", "Enabling module streams:", " postgresql                         10                                         ", "", "Transaction Summary", "================================================================================", "", "Complete!"]}

TASK [packages_el : Install Postgresql 10 [RHEL8]] **********************************************
ok: [ec2-xxx.eu-west-2.compute.amazonaws.com] => {"changed": false, "msg": "Nothing to do", "rc": 0, "results": []}

TASK [packages_el : Upgrade Postgres 10 [RHEL 8]] ***********************************************
fatal: [ec2-xxx.eu-west-2.compute.amazonaws.com]: FAILED! => {"changed": true, "cmd": ["/bin/postgresql-setup", "--upgrade", "--upgrade-from=postgresql-96"], "delta": "0:00:00.109436", "end": "2019-11-20 10:50:39.957904", "msg": "non-zero return code", "rc": 1, "start": "2019-11-20 10:50:39.848468", "stderr": "ERROR: Cannot upgrade because the database in /var/lib/pgsql/data is of\n       version 10 but it should be 9.6", "stderr_lines": ["ERROR: Cannot upgrade because the database in /var/lib/pgsql/data is of", "       version 10 but it should be 9.6"], "stdout": "", "stdout_lines": []}

##### ADDITIONAL INFORMATION

Inventory file contents (sensitive info masked):

[tower]
ec2-xxa.eu-west-2.compute.amazonaws.com
ec2-xxb.eu-west-2.compute.amazonaws.com
ec2-xxc.eu-west-2.compute.amazonaws.com
[database]
ec2-xxx.eu-west-2.compute.amazonaws.com
[all:vars]
ansible_become=true
ansible_user=ec2-user
ansible_ssh_private_key_file=xxx
admin_password='xxx'
pg_host='ec2-xxx.eu-west-2.compute.amazonaws.com'
pg_port='5432'
pg_database='xxx'
pg_username='xxx'
pg_password='xxx'
rabbitmq_port=5672
rabbitmq_vhost=xxx
rabbitmq_username=xxx
rabbitmq_password=xxx
rabbitmq_cookie=xxx
rabbitmq_use_long_name=true

Workaround on database node:
yum -y remove postgres\*
cd /var/lib && rm -rf pgsql
Then re-run setup.sh again

**Comment by shanemcd** [comment]
Hi. This is an issue tracker for the upstream project AWX, not Tower. Please contact support in the future. The issue you found here will be fixed in 3.6.1, out very soon.
